include/unifex/when_all.hpp

```diff
@@ -292,19 +304,39 @@ class _sender<Senders...>::type {
   explicit type(Senders2&&... senders)
     : senders_((Senders2 &&) senders...) {}
 
-  template <typename Receiver>
-  operation<Receiver, Senders...> connect(Receiver&& receiver) && {
+  template <
+    typename CPO,
+    typename Sender,
+    typename Receiver,
+    std::enable_if_t<
+      std::conjunction_v<
+        std::is_same<CPO, tag_t<unifex::connect>>,
+        std::is_same<Sender, type>,
+        when_all_connectable<std::remove_cvref_t<Receiver>, Senders...>>, int> = 0>
```

Comment:
It would be nice to eliminate the `conjunction` here as well. Is short-circuiting needed here?

Reply:
We want to avoid evaluating the when_all_connectable<> if the first two don't match, but we can probably evaluate the first two together.